contract-sdk/types/src/address.rs

```diff
@@ -40,6 +40,20 @@ impl Address {
         Ok(Self(a))
     }
 
+    /// Tries to create a new address from Bech32-encoded string.
+    pub fn from_bech32(data: &str) -> Result<Self, Error> {
+        let (hrp, data, variant) = bech32::decode(data).map_err(|_| Error::MalformedAddress)?;
+        if hrp != ADDRESS_BECH32_HRP {
+            return Err(Error::MalformedAddress);
```

Comment:
Would be great to have tests for these error cases.

Reply:
Added, also to the `runtime-sdk` address module